app/models/registration.rb

```diff
@@ -9,6 +9,7 @@ class Registration < ApplicationRecord
 
   belongs_to :event
   belongs_to :discount_code, counter_cache: true, optional: true
+  belongs_to :user, optional: true
```

Comment:
@willblake01 -- this is important. I recommend doing some googling on why the `optional` param is necessary here.

Reply:
👍